src/corelibs/U2Formats/src/PDBFormat.cpp

```diff
@@ -265,6 +268,17 @@ void PDBFormat::PDBParser::parseBioStruct3D(BioStruct3D& biostruct, U2OpStatus&
     updateSecStructChainIndexes(biostruct);
 }
 
+void PDBFormat::PDBParser::checkIdsOrder(QList<int>& idSequence, U2OpStatus& ti) {
+    QList<int> sortedIdSequence(idSequence);
+    std::sort(sortedIdSequence.begin(), sortedIdSequence.end());
+    if (sortedIdSequence != idSequence) {
+        ti.addWarning(PDBFormat::tr("The atoms in the file are not located according to the order of their identifiers."
```

Comment:
Is it OK to open such PDB files? Why don't we fail and mark the document as broken? Are there public sources (databases) that provide such broken files and what do we expect from user after showing the message in the log?

Reply:
There are no such files among ~30 PDBs in our repositories.
According to PDB format documentation http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM this field described as ''serial number", with no other description. As i think value supposed to follow natural order starting with 1.
According to our code this id used as key in map and it should not cause crash - we do not rely on their order, we wiil construct molecule in any case.
I think its not a criminal to open such files and show what we got, especially user already warned in log about wrong order in file.
It is up to you to decide which solution to apply.

Reply:
Do you think the user will understand the details in the message in order to make a decision?

If the wrong order can break anything in UGENE algorithms/rendering: we should not allow opening this file or reading the broken models at all.

If the wrong order is automatically recovered by UGENE the warning message should announce that "UGENE fixed the order of atoms..." and open the file.

Here is a source of hundreds of PDBs (if you need)
```
wget -m -nH --cut-dirs=6 -np --reject 'index.*' https://files.wwpdb.org/pub/pdb/data/structures/models/current/pdb/
```